### PR TITLE
Add fuzzer for the compact custom codec implementation from PR #6720

### PR DESCRIPTION
**Pull request by viniul**
This commit adds a fuzzing harness for the custom compact encoding/decoding
introduced in PR #6720.

### CODE REVIEW

File: primitives/npos-elections/fuzzer/src/compact.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                        )
+                        .unwrap();
+                    // And it should be equal to the original decoded object (resulting from directly
+                    // decoding fuzzer_data):
+                    assert_eq!(decoded, decoded2);
+                }
+                Err(_e) => {
+                    // Can happen, not every random sequence of bytes an be decoded as InnerTestSolutionCompact
+                }
+            }
```
Comment (bkchr) [inline primitives/npos-elections/fuzzer/src/compact.rs:36]:
```suggestion
            // Ignore errors as not every random sequence of bytes can be decoded as InnerTestSolutionCompact
            if let Ok(decoded) = result_decoded {
                // Decoding works, let's re-encode it and compare results.
                let reencoded: std::vec::Vec<u8> = decoded.encode();
                // The reencoded value may or may not be equal to the original fuzzer output. However, the
                // original decoder should be optimal (in the sense that there is no shorter encoding of
                // the same object). So let's see if the fuzzer can find something shorter:
                if fuzzer_data.len() < reencoded.len() {
                    panic!("fuzzer_data.len() < reencoded.len()");
                }
                // The reencoded value should definitely be decodable (if unwrap() fails that is a valid
                // panic/finding for the fuzzer):
                let decoded2: InnerTestSolutionCompact =
                    <InnerTestSolutionCompact as codec::Decode>::decode(
                        &mut reencoded.as_slice(),
                    ).unwrap();
                // And it should be equal to the original decoded object (resulting from directly
                // decoding fuzzer_data):
                assert_eq!(decoded, decoded2);
            }
```

**Review by bkchr — COMMENTED** [review]
We indent using tabs, could you please convert the file?

**Comment by bkchr** [comment]
And you also need to commit the `Cargo.lock` file

**Comment by viniul** [comment]
> And you also need to commit the `Cargo.lock` file

Done

**Comment by bkchr** [comment]
@viniul could you also change the indentation to tabs? :) Then everything is fine!

**Comment by viniul** [comment]
> @viniul could you also change the indentation to tabs? :) Then everything is fine!

Should be done now. Thanks for reviewing!

**Review by gavofyork — APPROVED** [review]

**Comment by gavofyork** [comment]
bot merge

**Pull request merged**